### DISCUSSION TITLE
sorted window list at R-Mouse-Button

### DISCUSSION THREAD

**Discussion by nnako** [freeplane/freeplane#397] [Ideas]
In order to keep an overview of all currently opened maps and being able to quickly jump between them, the map's context menu can be opened by pressing

`<RCLICK>` on an empty space within the current map

It shows the a list of opened Freeplane maps, and further down some map-specific configurations. Now, when there are many opened maps (so that even the map's tab title strings become strongly abbrediated), the list of open maps from the map's context menu could still be of great help.

But, as currently the list does not seem to be sorted in any particular way (at least not alphabetically), this makes it difficult to quickly find the desired map title to be activated.

**Would it be possible to sort the list of open maps before displaying their titles within the map's context menu?** For users who'd like to stick to the existing (unsorted) map list, an appropriate settings item could be provided.

What do you think?

**Comment by macmarrum** [comment]
I can see that the order in the context menu reflects the order of the tabs and this is fine with me.

![order_or_tabs](https://user-images.githubusercontent.com/7197131/169837179-566938e4-4670-40d7-adcc-8a3ab1798dda.png)

**Comment by nnako** [comment]
Hi @macmarrum ,

what if you have about 30 open maps and you have to switch between them alot...

![image](https://user-images.githubusercontent.com/5845602/169960358-5d2302af-94a2-4053-8e9c-d1724a997374.png)

Some maps are open even on the same tab / window, so that the tab title will not be capable of showing titles in proper length. And, e.g. the two maps starting with "project_..." in the screenshot above reside even next to each other on the same window. But they are located far away from each other within the list. The current ordering mechanism seems to be a mixture of `location on tab / window` and `recent modification time`.

An alphabetical order would definitely be helpful in this case, I think.

**Reply by macmarrum** [reply]
> An alphabetical order would be helpful in this case.

@nnako I can clearly see that it would be helpful for you.

All I am saying is that the approach to sorting files in the context menu is known. The order is set manually by dragging the tabs around. And that this is enough for my needs.

Therefore, to answer your question, if I had ~30 tabs open and wanted to have the context-menu file names sorted alphabetically to have a better overview, I would move the tabs around.

**Reply by nnako** [reply]
@macmarrum 
> Therefore, to answer your question, if I had ~30 tabs open and wanted to have the context-menu file names sorted alphabetically to have a better overview, I would move the tabs around.

sorting the tabs might work for a concept of one-tab-per-window, but multiple tabs on a single window don't seem to follow the tab-order-scheme mentioned. And, I would rather re-order my tabs depending on my current task (some maps will be closed and some maps will (re-)enter the scene)... not just in order to produce an alphabetical order within the map's context menu.

Generally I think, there is still space for solutions aiming for optimized handling of growing information databases. This trend will ultimately lead to `big maps`, `multiple interconnected maps`, `time-consuming searches` ... , I think.

**Reply by euu2021** [reply]
I don't know if you know about tab grouping, as it's a non-obvious feature. It could be useful in your situation (edit: I mean, useful for tab management in general, and not for the specific problem of alphabetic order of the tabs). I  wrote about it [here](https://sourceforge.net/p/freeplane/discussion/758437/thread/9d5be10a28/?limit=25#42fd)

**Reply by macmarrum** [reply]
Thanks for the refresher, @eu2021. I keep forgetting about this feature 😊

**Comment by sifran-github** [comment]
I agree with @macmarrum that the main behavior that I like is the current one, where the tabs order is sorted by the user.

Generally, also the nodes in mind maps themselves are sorted manually by the user and not automatically by alphabet, and that is good.

I guess what you are looking for (sorted automatically) can be implement as an add on.

**Comment by dpolivaev** [comment]
For me the idea of sorting mind maps alphabetically makes sense. I have already implemented it (but not created a preview yet)
I want you to test it before hiding it behind an option.

**Reply by macmarrum** [reply]
It looks good to me ✅ 

![order_of_tabs_alphabetically](https://user-images.githubusercontent.com/7197131/170038148-26babf94-b9d5-4be6-bbf7-a3c3278d5f38.png)

**Reply by nnako** [reply]
@dpolivaev excellent!
